plugins/woocommerce/includes/gateways/paypal/class-wc-gateway-paypal.php

```diff
@@ -696,25 +668,70 @@ public function has_paypal_orders() {
 	}
 
 	/**
-	 * Checks if the gateway should use Orders v2 API.
+	 * Check if the gateway should use Orders v2 API.
 	 *
 	 * @return bool
 	 */
-	protected function should_use_orders_v2() {
-		// phpcs:ignore Generic.Commenting.Todo.TaskFound
-		// TODO: We expect this flag to be true if the merchant can be migrated,
-		// i.e. does not need PayPal API keys, and they have accepted the ToS.
-
+	public function should_use_orders_v2() {
```

Comment:
@Mayisha FYI that I'm moving this check from the helper to the gateway class, as it needs the gateway instance to perform the Transact checks.

I added a filter to handle the legacy settings logic.